### PR TITLE
Allow failsafe gimbal action

### PR DESCRIPTION
**Pull request by dayjaby**
**Describe problem solved by this pull request**
It is already possible to set a gimbal orientation during failsafe when the gimbal is controlled via PWM. With this PR, it becomes possible to let any kind of gimbal perform a failsafe action. At the moment, the only action implemented is "look 45 degrees upwards" in order to prevent damage on the camera.

**Describe your solution**
If failsafe is triggered, we send a CMD_DO_MOUNT_CONTROL, which is usually consumed by the vmount module.

If the gimbal is controlled by RC, this implementation has no effect.

### CODE REVIEW

**Comment by dagar** [comment]
What's the actual gimbal architecture here? I'm wondering if `vmount` might be a more appropriate home for this kind of thing. 

https://github.com/PX4/Firmware/blob/master/src/modules/vmount/input_mavlink.cpp

**Comment by dayjaby** [comment]
Moved it to vmount. Still having the issue that the implementation has no effect if gimbal is controlled by RC.

File: src/modules/vmount/input.cpp [hunk 1]
```diff
@@ -60,6 +60,27 @@ int InputBase::update(unsigned int timeout_ms, ControlData **control_data, bool
 		return 0;
 	}
 
+	if (_failsafe_action == 1 && !_failsafe_triggered && _actuator_armed_sub.updated()) {
```
Comment (LorenzMeier) [inline src/modules/vmount/input.cpp:63]: Could you please introduce an ENUM for the failsafe action and use that here?

Comment (dayjaby) [reply src/modules/vmount/input.cpp]: Done.

**Comment by julianoes** [comment]
I think this change is fine as such but I'm not sure if it will survive for long given that we will move to the new gimbal protocol and many things around vmount will change.
https://mavlink.io/en/services/gimbal_v2.html

**Comment by julianoes** [comment]
Just looking at this again and wondering if the "failsafe action" of a gimbal should not just be "retract" instead? And "retract" would then map to whatever position is safe for a gimbal.

**Comment by dayjaby** [comment]
@julianoes sounds good to me. Should I completely get rid of the gimbal v1 way of setting this 45 degrees pitch?

**Comment by julianoes** [comment]
I'd say so, yes.

**Comment by dayjaby** [comment]
Out-of-date since gimbalv2.

**Pull request closed without merging**